include/meta/meta.hpp

```diff
@@ -874,8 +880,9 @@ namespace meta
     /// \endcond
 
     /// Return a new \c meta::list constructed by doing a right fold of the list
-    /// \p List using binary Metafunction Class \c Fun and initial state \c
-    /// State.
+    /// \p List using binary Metafunction Class \p Fun and initial state \p
+    /// State. That is, the \c State_N for the list element \c A_N is computed
+    /// by `Fun(A_N, State_N+1) -> State_N`.
```

Comment:
I always have to look up what `foldl` and `foldr` do to pass an appropriate binary metafunction, I hope I didn't make it worse.